File: ElenaMartiniKreationen/FrontEnd/EcommerceApp/src/app/services/product/response/product-dto-response.ts
export class ProductDtoResponse {
  id: number;
  productId: number; // Alias para el ID del producto
  name: string;
  description: string;
  category: string;
  price: number;
  stock: number;
  imageUrl: string;
  quantity: number;
  cartItemId?: number; 
  cartId?: number | undefined;

  constructor(data: any) {
    this.id = data.id;
    this.productId = data.id; // Mapea id a productId
    this.name = data.name;
    this.description = data.description;
    this.price = data.price;
    this.stock = data.stock;
    this.category = data.category;
    this.imageUrl = data.imageUrl;
    this.quantity = data.quantity || 1;
    this.cartItemId = data.cartItemId;
    this.cartId = data.cartId;
  }
}

File: BackEnd/ElenaMartiniKreationen.Server/Response/ProductDtoResponse.cs
﻿namespace ElenaMartiniKreationen.Server.Response
{
    public class ProductDtoResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = default!;
        public string Description { get; set; } = default!;
        public decimal? Price { get; set; }
        public int Stock { get; set;}
        public string ImageUrl { get; set; } = default!;
    }
}
